samples/nrf9160/modem_shell/src/ltelc/ltelc_settings.c

```diff
@@ -0,0 +1,794 @@
+
+	char defcontauth_uname_str[LTELC_SETT_DEFCONTAUTH_MAX_UNAME_STR_LEN + 1];
+	char defcontauth_pword_str[LTELC_SETT_DEFCONTAUTH_MAX_PWORD_STR_LEN + 1];
+	bool defcontauth_enabled;
+
+	enum lte_lc_system_mode sysmode;
+	enum lte_lc_system_mode_preference sysmode_lte_preference;
+	enum pdn_fam pdn_family;
+	enum pdn_auth defcontauth_prot;
+
+	/* Note: if adding more memory slots, remember also update
+	 * LTELC_SETT_NMODEAT_MEM_SLOT_INDEX_START/END accordingly.
+	 */
+	char normal_mode_at_cmd_str_1
+	[CONFIG_MOSH_LTELC_SETT_NORMAL_MODE_AT_CMD_STR_LEN + 1];
+	char normal_mode_at_cmd_str_2
+	[CONFIG_MOSH_LTELC_SETT_NORMAL_MODE_AT_CMD_STR_LEN + 1];
+	char normal_mode_at_cmd_str_3
+	[CONFIG_MOSH_LTELC_SETT_NORMAL_MODE_AT_CMD_STR_LEN + 1];
+
+	bool normal_mode_autoconn_enabled;
+};
+static struct ltelc_sett_t ltelc_settings;
+
+/* ****************************************************************************/
+
+/**@brief Callback when settings_load() is called. */
+static int ltelc_sett_handler(const char *key, size_t len,
+			      settings_read_cb read_cb, void *cb_arg)
+{
+	int err;
+
+	if (strcmp(key, LTELC_SETT_DEFCONT_ENABLED) == 0) {
+		err = read_cb(cb_arg, &ltelc_settings.defcont_enabled,
+			      sizeof(ltelc_settings.defcont_enabled));
+		if (err < 0) {
+			shell_error(uart_shell,
+				    "Failed to read defcont enabled, error: %d",
+				    err);
+			return err;
+		}
+		return 0;
+	} else if (strcmp(key, LTELC_SETT_DEFCONT_APN_KEY) == 0) {
+		err = read_cb(cb_arg, &ltelc_settings.defcont_apn_str,
+			      sizeof(ltelc_settings.defcont_apn_str));
+		if (err < 0) {
+			shell_error(uart_shell,
+				    "Failed to read defcont APN, error: %d",
+				    err);
+			return err;
+		}
+		return 0;
+	} else if (strcmp(key, LTELC_SETT_DEFCONT_IP_FAMILY_KEY) == 0) {
+		err = read_cb(cb_arg, &ltelc_settings.pdn_family,
+			      sizeof(ltelc_settings.pdn_family));
+		if (err < 0) {
+			shell_error(
+				uart_shell,
+				"Failed to read defcont PDN family, error: %d",
+				err);
+			return err;
+		}
+		return 0;
+	}
+	if (strcmp(key, LTELC_SETT_DEFCONTAUTH_ENABLED) == 0) {
+		err = read_cb(cb_arg, &ltelc_settings.defcontauth_enabled,
+			      sizeof(ltelc_settings.defcontauth_enabled));
+		if (err < 0) {
+			shell_error(
+				uart_shell,
+				"Failed to read defcontauth enabled, error: %d",
+				err);
+			return err;
+		}
+		return 0;
+	} else if (strcmp(key, LTELC_SETT_DEFCONTAUTH_USERNAME_KEY) == 0) {
+		err = read_cb(cb_arg, &ltelc_settings.defcontauth_uname_str,
+			      sizeof(ltelc_settings.defcontauth_uname_str));
+		if (err < 0) {
+			shell_error(
+				uart_shell,
+				"Failed to read defcontauth username, error: %d",
+				err);
+			return err;
+		}
+		return 0;
+	} else if (strcmp(key, LTELC_SETT_DEFCONTAUTH_PASSWORD_KEY) == 0) {
+		err = read_cb(cb_arg, &ltelc_settings.defcontauth_pword_str,
+			      sizeof(ltelc_settings.defcontauth_pword_str));
+		if (err < 0) {
+			printf("Failed to read defcontauth password, error: %d",
```

Comment:
Why `printf` all of sudden?

Reply:
I've changed all these printf's to shell_print/shell_error throughout MOSH. However, there are some that are call from main() which are left as printk as shell is not ready to print necessarily.